test/StaticAnalysis/PluginRetrievalTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Filter\StaticAnalysis;
+
+use Laminas\Filter\FilterPluginManager;
+use Laminas\Filter\StringToUpper;
+
+final class PluginRetrievalTest
+{
+    private FilterPluginManager $pluginManager;
+
+    public function __construct(FilterPluginManager $pluginManager)
+    {
+        $this->pluginManager = $pluginManager;
+    }
+
+    /** @return mixed */
+    public function filterSomethingWithAKnownFilterClass(string $value)
+    {
+        $plugin = $this->pluginManager->get(StringToUpper::class);
```

Comment:
Assuming type inference on the plugin manager is improved, couldn't we infer `string` as a return value here?

`StringToUpper` would return `string` if the input was a `string` :thinking:

Reply:
Got it. This would all be a whole lot easier if we only ever returned instances of FilterInterface - could we wrap all callables in the `Callback`Filter in a minor or would that require a major?